### PR TITLE
JAX docs tutorial: JAX internals 301 - The jaxpr language

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp 

https://github.com/google/jax/issues/18585

### CODE REVIEW

File: docs/tutorials/jaxpr.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+```{code-cell}
+from jax import jit
+
+def func12(arg):
+  @jit
+  def inner(x):
+    return x + arg * jnp.ones(1)  # Include a constant in the inner function.
+  return arg + inner(arg - 2.)
+
+print(make_jaxpr(func12)(1.))  # doctest:+ELLIPSIS
```
Comment (jakevdp) [inline docs/tutorials/jaxpr.md:289]: `doctest` doesn't apply in `code-cell` blocks.

File: docs/tutorials/jaxpr.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+The `linear` parameter describes for each of the input variables whether they are guaranteed to be used linearly in the body. Once the `scan` goes through linearization, more arguments will be linear.
+
+The `scan` primitive takes 4 arguments: `b 0.0 a c`, of which:
+
+- One is the free variable for the body
+- One is the initial value of the carry
+- The next 2 are the arrays over which the scan operates
+
+
+### `xla_call` primitive
```
Comment (jakevdp) [inline docs/tutorials/jaxpr.md:276]: `xla_call` has been replaced by `pjit`

File: docs/tutorials/jaxpr.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+def func12(arg):
+  @jit
+  def inner(x):
+    return x + arg * jnp.ones(1)  # Include a constant in the inner function.
+  return arg + inner(arg - 2.)
+
+print(make_jaxpr(func12)(1.))  # doctest:+ELLIPSIS
+```
+
+### `xla_pmap` primitive
```
Comment (jakevdp) [inline docs/tutorials/jaxpr.md:292]: This is deprecated... remove, or maybe replace with a `shard_map` example?

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: docs/tutorials/jaxpr.md (hunk 2 above)
Comment (8bitmp3) [reply docs/tutorials/jaxpr.md:276]: `(p)jit`

File: docs/tutorials/jaxpr.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+- `Expr+` are one or more atomic expressions, each either a variable or a literal constant. A special variable `unitvar` or literal `unit`, printed as `*`, represents a value that is not needed in the rest of the computation and has been elided. That is, units are just placeholders.
+- `Param*` are zero or more named parameters to the primitive, printed in square brackets. Each parameter is shown as `Name = Value`.
+
+Most jaxpr primitives are first-order (they take just one or more Expr as arguments):
+
+```
+Primitive := add | sub | sin | mul | ...
+```
+
+The jaxpr primitives are documented in the {mod}`jax.lax` module.
```
Comment (jakevdp) [inline docs/tutorials/jaxpr.md:88]: This is no longer totally true; I'd say "The most common jaxpr primitives are documented..."

Comment (8bitmp3) [reply docs/tutorials/jaxpr.md:88]: On it.

**Review by jakevdp — APPROVED** [review]

**Comment by 8bitmp3** [comment]
:fire: :rocket:

**Pull request merged**